src/tcgwars/logic/impl/gen7/UnbrokenBonds.groovy

```diff
@@ -4256,6 +4256,13 @@ public enum UnbrokenBonds implements LogicCardInfo {
             bg.em().storeObject("KOGA_S_TRAP_TURN", bg.turnCount)
           }
           playRequirement{
+            def defendingConfused = opp.active.isSPC(CONFUSED)
+            def defendingPoisoned = opp.active.isSPC(POISONED)
+            if (defendingPoisoned){
+              //Check for multi-counter poison; this only detects active increasers.
+              def extraPoisonCount = bg.em().retrieveObject("extra_poison_counter_"+opp.active.hashCode()) ?: 0
+            }
+            assert !defendingConfused || !defendingPoisoned || extraPoisonCount : "Your opponent's Active Pokémon is already Confused and Poisoned"
```

Comment:
Wouldn't this block the situation where the you want to use Koga's Trap when the opponent's active is either only confused or only poisoned?

Reply:
As long as one of the three conditions is true, it should allow to use the card. I'm on doubt about extraPoisonCount being reset when poison is removed though, will check in a bit.

But being only poisoned/confused should pass this assert.

Reply:
Ok, there seems to be an EE on Vileplume Delta. Pushed a fix for it and Koga, should work fine now (will push a fix if it keeps erroring after merge).